File: pickerkit/src/main/java/com/dewarder/pickerkit/provider/PickerDataProvider.java
package com.dewarder.pickerkit.provider;

import java.util.Collection;

public interface PickerDataProvider<T> {

    void request(Callback<T> callback);

    interface Callback<T> {

        void onNext(Collection<T> data);

        void onComplete();

        void onError(Throwable throwable);
    }
}

File: pickerkit/src/main/java/com/dewarder/pickerkit/provider/PickerDataProvider.kt
package com.dewarder.pickerkit.provider

interface PickerDataProvider<out T> {

    fun request(callback: Callback<T>)

    interface Callback<in T> {

        fun onNext(data: Collection<T>)

        fun onComplete()

        fun onError(throwable: Throwable)
    }
}
